### DISCUSSION TITLE
Clarification about `LazyFile` event

### DISCUSSION THREAD

**Discussion by dpetka2001** [LazyVim/LazyVim#1583] [Q&A] [answered]
Hi @folke, 
I saw today the introduction of `LazyFile` event. Would you mind clarifying a bit when that should be used in a user's personal spec? When exactly is this event happening? After `BufReadPost` for example?

**Comment by folke** [comment, marked as answer]
It's a shortcut for `event = { "BufReadPost", "BufWritePost", "BufNewFile" }`, but defers (and re-triggers) the event to make sure the ui isn't blocked for initial rendering.

Previously when you opened a file directly like `nvim foo.lua`, then Neovim triggers `BufReadPost` events **before** `VimEnter`.
This is annoying, since in our case lsp-config, treesitter and others would block showing the actual file.

The new change renders the file as fast as possible and right after that loads any file-based plugins.
As a consequence,  the startuptime of `nvim foo.lua` will now be similar to just doing `nvim`.

I mainly added this as an internal event for LazyVim specs, but you're of course free to also use this in your own config.

**Reply by dpetka2001** [reply]
What kind of plugins would you suggest load with this event? I have some in my personal configuration like `Comment.nvim`, `vim-matchup` and a couple others that load on `{ "BufReadPost", "BufNewFile" }` events. should i move them to `LazyFile` instead. Is there a safe way to judge when to do this?

**Reply by folke** [reply]
Yep, should safe to move, but test it to make sure it still works ;)

**Reply by dpetka2001** [reply]
Tnx a lot for clarifying. Moved all of them and didn't notice any weird behavior for now. If I do, I'll open an appropriate issue.

**Reply by baggiponte** [reply]
Do you think this could be added to lazy.nvim like the "VeryLazy" event?

**Reply by folke** [reply]
No, because the `LazyFile` event is way more involved. People would use it for things they shouldn't and their configs would break.

On the LAzyVim side, I have control over how it's being used

**Reply by synic** [reply]
@folke I'm wondering if this is what i'm experiencing in my own config.

Sometimes, when opening a file from telescope (before lsp, treesitter, etc are loaded), the file is blank and read-only. Other files load just fine, presumably because plugins are now loaded.

If this is possibly my issue, do you have suggestions on how to solve it outside of LazyVim? 

Thank you

**Reply by Crypto-Spartan** [reply]
I use [lazy.nvim](https://github.com/folke/lazy.nvim) and I would love to incorporate the `LazyFile` event into my neovim config. **I completely understand** that I'm entering "_use at your own risk_" territory, especially with the disclaimer above:

>  People would use it for things they shouldn't and their configs would break.

I've been reading through the LazyVim code, and I found where the `LazyFile` event is registered: https://github.com/LazyVim/LazyVim/blob/d0c366e4d861b848bdc710696d5311dca2c6d540/lua/lazyvim/util/plugin.lua#L73-L79

This is very straightforward. However, I don't see the code anywhere that handles this:
> but defers (and re-triggers) the event to make sure the ui isn't blocked for initial rendering.

@folke could you point me in the direction of where that code is? Thank you for all of your hard work (:

**Reply by Ra2-IFV** [reply]
Same for me.
